### PR TITLE
Move checkbox-ce-oem snapcraft.yaml into contrib area (New)

### PR DESCRIPTION
**Pull request by zongminl**
## Description

There were snapcraft.yaml files hosted on LP Git for checkbox-ce-oem to be built and published by LP build infra, as of now checkbox-provider-ce-oem has been hosted in the contrib area of monorepo we should also host snapcraft.yaml files on GH so checkbox-ce-oem snap can be built through GH workflow

Note: This PR only moves snapcraft.yaml files into GH, there hasn't been any changes with regard to building latest checkbox-ce-oem source from contrib area

## Resolved issues

https://warthogs.atlassian.net/browse/OEMQA-4135

### CODE REVIEW

File: contrib/checkbox-provider-ce-oem/series_classic20/launchers/configure [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    msg = """
+checkbox-support not found!
+You need to connect this snap to the checkbox20 snap.
+
+You can do this with those commands:
+
+snap connect $SNAP_NAME:checkbox-runtime  checkbox20:checkbox-runtime
+snap connect $SNAP_NAME:provider-resource checkbox20:provider-resource
+snap connect $SNAP_NAME:provider-checkbox checkbox20:provider-checkbox
+snap connect $SNAP_NAME:provider-ipdt     checkbox20:provider-ipdt
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_classic20/launchers/configure:36]: Remove `provider-ipdt `
```suggestion
```

File: contrib/checkbox-provider-ce-oem/series_classic20/snap/hooks/configure [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    msg = """
+checkbox-support not found!
+You need to connect this snap to the checkbox22 snap.
+
+You can do this with those commands:
+
+snap connect $SNAP_NAME:checkbox-runtime  checkbox22:checkbox-runtime
+snap connect $SNAP_NAME:provider-resource checkbox22:provider-resource
+snap connect $SNAP_NAME:provider-checkbox checkbox22:provider-checkbox
+snap connect $SNAP_NAME:provider-ipdt     checkbox22:provider-ipdt
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_classic20/snap/hooks/configure:56]: Remove `provider-ipdt `
```suggestion
```

File: contrib/checkbox-provider-ce-oem/series_classic22/launchers/configure [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    msg = """
+checkbox-support not found!
+You need to connect this snap to the checkbox22 snap.
+
+You can do this with those commands:
+
+snap connect $SNAP_NAME:checkbox-runtime  checkbox22:checkbox-runtime
+snap connect $SNAP_NAME:provider-resource checkbox22:provider-resource
+snap connect $SNAP_NAME:provider-checkbox checkbox22:provider-checkbox
+snap connect $SNAP_NAME:provider-ipdt     checkbox22:provider-ipdt
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_classic22/launchers/configure:36]: Remove `provider-ipdt `
```suggestion
```

File: contrib/checkbox-provider-ce-oem/series_classic22/snap/hooks/configure [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    msg = """
+checkbox-support not found!
+You need to connect this snap to the checkbox22 snap.
+
+You can do this with those commands:
+
+snap connect $SNAP_NAME:checkbox-runtime  checkbox22:checkbox-runtime
+snap connect $SNAP_NAME:provider-resource checkbox22:provider-resource
+snap connect $SNAP_NAME:provider-checkbox checkbox22:provider-checkbox
+snap connect $SNAP_NAME:provider-ipdt     checkbox22:provider-ipdt
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_classic22/snap/hooks/configure:56]: Remove `provider-ipdt `
```suggestion
```

File: contrib/checkbox-provider-ce-oem/series_uc22/launchers/configure [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    msg = """
+checkbox-support not found!
+You need to connect this snap to the checkbox22 snap.
+
+You can do this with those commands:
+
+snap connect $SNAP_NAME:checkbox-runtime  checkbox22:checkbox-runtime
+snap connect $SNAP_NAME:provider-resource checkbox22:provider-resource
+snap connect $SNAP_NAME:provider-checkbox checkbox22:provider-checkbox
+snap connect $SNAP_NAME:provider-ipdt     checkbox22:provider-ipdt
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_uc22/launchers/configure:36]: Remove `provider-ipdt `
```suggestion
```

File: contrib/checkbox-provider-ce-oem/series_classic20/snap/snapcraft.yaml [hunk 6]
```diff
@@ -0,0 +1,100 @@
+name: checkbox-ce-oem
+summary: Checkbox CE OEM test runner and public providers for 20.04 classic
+description: "Checkbox CE OEM test runner and public providers for 20.04 classic"
+confinement: classic
+grade: stable
+
+version: '0.12-focal'
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_classic20/snap/snapcraft.yaml:7]: How about starting the version from `1.0`?
```suggestion
version: '1.0-focal'
```

File: contrib/checkbox-provider-ce-oem/series_classic22/snap/snapcraft.yaml [hunk 7]
```diff
@@ -0,0 +1,100 @@
+name: checkbox-ce-oem
+summary: Checkbox CE OEM test runner and public providers for 22.04 classic
+description: "Checkbox CE OEM test runner and public providers for 22.04 classic"
+confinement: classic
+grade: stable
+
+version: '0.12-jammy'
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_classic22/snap/snapcraft.yaml:7]: How about starting the version from `1.0`?
```suggestion
version: '1.0-jammy'
```

File: contrib/checkbox-provider-ce-oem/series_uc20/snap/snapcraft.yaml [hunk 8]
```diff
@@ -0,0 +1,57 @@
+name: checkbox-ce-oem
+summary: Checkbox CE OEM and public providers
+description: "Checkbox CE OEM and public providers"
+confinement: strict
+grade: stable
+
+version: '0.12-focal'
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_uc20/snap/snapcraft.yaml:7]: How about starting the version from `1.0`?
```suggestion
version: '1.0-focal'
```

File: contrib/checkbox-provider-ce-oem/series_uc22/snap/snapcraft.yaml [hunk 9]
```diff
@@ -0,0 +1,136 @@
+name: checkbox-ce-oem
+summary: Checkbox CE OEM test runner and public providers
+description: "Checkbox CE OEM test runner and public providers"
+confinement: strict
+grade: stable
+
+version: '0.12-jammy'
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_uc22/snap/snapcraft.yaml:7]: How about starting the version from 1.0?
```suggestion
version: '1.0-jammy'
```

File: contrib/checkbox-provider-ce-oem/series_classic22/launchers/wrapper_local [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    echo "You can do this with this command:"
+    echo "snap install checkbox22"
+    exit 1
+fi
+
+export LC_ALL=C.UTF-8
+PERL_VERSION=$(perl -e '$^V=~/^v(\d+\.\d+)/;print $1')
+export PERL5LIB="$PERL5LIB:$SNAP/usr/lib/$ARCH/perl/$PERL_VERSION:$SNAP/usr/lib/$ARCH/perl5/$PERL_VERSION:$SNAP/usr/share/perl/$PERL_VERSION:$SNAP/usr/share/perl5"
+export GI_TYPELIB_PATH=$SNAP/usr/lib/girepository-1.0:$SNAP/usr/lib/$ARCH/girepository-1.0
+#export PATH="$PATH:$SNAP/usr/bin:$SNAP/usr/sbin:$SNAP/sbin:/snap/bin:/snap/checkbox-ce-oem/current/usr/bin/:/snap/checkbox-ce-oem/current/usr/sbin"
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_classic22/launchers/wrapper_local:34]: Remove useless `export`
```suggestion
```

File: contrib/checkbox-provider-ce-oem/series_classic22/launchers/wrapper_local [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+export PATH="$PATH:$SNAP/usr/bin:$SNAP/usr/sbin:$SNAP/sbin:/snap/bin"
+export PYTHONPATH="$SNAP/usr/lib/python3/dist-packages:$PYTHONPATH"
+#export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/snap/checkbox-ce-oem/current/usr/lib/aarch64-linux-gnu"
```
Comment (baconYao) [inline contrib/checkbox-provider-ce-oem/series_classic22/launchers/wrapper_local:37]: Remove useless `export`
```suggestion
```

**Review by baconYao — CHANGES_REQUESTED** [review]
@zongminl Please take a look at my suggestions,

**Review by rickwu666666 — DISMISSED** [review]
LGTM

**Review by LiaoU3 — APPROVED** [review]
+1

**Review by baconYao — APPROVED** [review]
+1

**Pull request merged**